FuelGaugeReader/FuelGaugeReader.ino

```diff
@@ -0,0 +1,38 @@
+#define USE_USBCON
+#include <ros.h>
+#include <std_msgs/Int64.h>
+#include <Arduino.h>
+std_msgs::Int64 fuel_level_percent_full;
+ros::Publisher pub_fuel_level("fuel_level_percent_full", &fuel_level_percent_full);
+ros::NodeHandle nh;
+
+const long LOOP_DELAY = 200;
+const int FUEL_GAUGE_READ_PIN = 61;
+const double VOLTAGE_ON_EMPTY = 2.45;
+const double VOLTAGE_ON_FULL = 3.49;
+const double GAUGE_INPUT_VOLTAGE = 5;
+const double ADC_RANGE = 1024.;
+
+void setup()
+{
+    nh.initNode();
+    Serial.begin(9600);
+    nh.getHardware()->setBaud(9600);
+    nh.advertise(pub_fuel_level);
+}
+
+void loop() {
+    static unsigned long current_time = millis();
+    static unsigned long last_fuel_level_pub_time = current_time;
+    current_time = millis();
+    if (current_time - last_fuel_level_pub_time >= LOOP_DELAY) {
+        // this exists so the calculation isn't on one big line
+        double percentage_calculation = analogRead(FUEL_GAUGE_READ_PIN) * GAUGE_INPUT_VOLTAGE / ADC_RANGE;
```

Comment:
If we are going to break it up lets break it up and call it what it is. This calculation actually is the voltage at the pin of the ADC on the MCU
```suggestion
        double measured_voltage = analogRead(FUEL_GAUGE_READ_PIN) * GAUGE_INPUT_VOLTAGE / ADC_RANGE;
```

Reply:
This has been renamed!